src/extensions/CanvasWidget/CanvasWidgetLayer.ts

```diff
@@ -260,29 +268,65 @@ export class CanvasWidgetLayer<LayerProps extends BaseLayerProps, State extends
     }
 }
 
-export const useCanvasWidgetLayer = <LayerProps extends BaseLayerProps, LayerState extends Object>(createLayer: () => CanvasWidgetLayer<LayerProps, LayerState>): CanvasWidgetLayer<LayerProps, LayerState> | null => {
+export const useLayer = <LayerProps extends BaseLayerProps, LayerState extends Object>(createLayer: () => CanvasWidgetLayer<LayerProps, LayerState>, layerProps?: LayerProps): CanvasWidgetLayer<LayerProps, LayerState> | null => {
     const [layer, setLayer] = useState<CanvasWidgetLayer<LayerProps, LayerState> | null>(null)
     useEffect(() => {
         if (layer === null) {
             setLayer(createLayer())
         }
     }, [layer, setLayer, createLayer])
+    if ((layer) && (layerProps)) {
+        layer.setProps(layerProps)
+    }
     return layer
 }
 
-export const useCanvasWidgetLayers = <LayerProps extends BaseLayerProps>(layerList: (CanvasWidgetLayer<LayerProps, any> | null)[]): CanvasWidgetLayer<LayerProps, any>[] | null => {
-    const [layers, setLayers] = useState<CanvasWidgetLayer<LayerProps, Object>[] | null>(null)
-    useEffect(() => {
-        if (layers === null) {
-            if (layerList.filter(L => (L === null)).length === 0) {
-                const layerList2: CanvasWidgetLayer<LayerProps, any>[] = []
-                layerList.forEach(L => {
-                    if (L === null) throw Error('Unexpected null layer')
-                    layerList2.push(L)
-                })
-                setLayers(layerList2)
-            }
+const listsMatch = (list1: any[], list2: any[]) => {
+    if (list1.length !== list2.length) return false
+    for (let i = 0; i < list1.length; i++) {
+        if (list1[i] !== list2[i]) return false
+    }
+    return true
+}
+
+export const useLayers = (layers: (CanvasWidgetLayer<any, any> | null)[]) => {
```

Comment:
useLayers is now somewhat different than the previous useCanvasWidgetLayers. It basically returns a persistent list of layers that only changes if the individual entries change. This is important to avoid CanvasWidget thinking that the layers props has changed even when the entries of the list have remained the same.